WordPress/openverse
# Spacing improvements in All content's <header> area

## Fixes



Fixes #2079 by @fcoveram 

## Description


This PR adds the following spacing and text style changes from the #2079 to improve the layout:
- Reduce the line height of Heading 1 text style from 1.3 to 1.
- Apply Heading 2 text style to `<h1>` on md breakpoint.
- Change the paddings of `<header>` element where h1 is placed to be 8px for top and 24px for bottom.
- Hide the <header> on xs and sm as the element adds unnecessary space.

## Testing Instructions

Check that the "All content" header area matches the designs [all layouts](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=4174-10738&t=da4IgBUXw0xPrjhT-4) in Views page in the Design Library for reference.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
